### PR TITLE
Add tinytga crate

### LINKED ISSUE #66: Tracking issue: TGA support
Use [nom](https://crates.io/crates/nom) in `no_std` mode to parse both uncompressed and RLE (Run Length Encoding) compressed TGA images. It should produce an iterator to stay true to the minimal-alloc philosophy of embedded_graphics.

Some work has been started [here](https://github.com/jamwaffles/embedded-graphics/tree/tga).

### PR DESCRIPTION
**Pull request by jamwaffles**
Closes #66 

TODO

* [x] Add a few more tests with images exported from other programs
* [x] DRY up the iterator code and see if it can be optimised some

### CODE REVIEW

File: embedded-graphics/src/image/image_tga.rs [hunk 1]
```diff
@@ -0,0 +1,170 @@
+use super::super::drawable::*;
+use super::super::transform::*;
+use super::ImageFile;
+use crate::coord::{Coord, ToUnsigned};
+use crate::pixelcolor::PixelColor;
+use crate::unsignedcoord::{ToSigned, UnsignedCoord};
+use core::marker::PhantomData;
+use tinytga::Tga;
+
+/// BMP format image
```
Comment (rfuest) [inline embedded-graphics/src/image/image_tga.rs]: Should be: TGA format image

File: embedded-graphics/src/image/image_tga.rs [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+            im: self,
+            image_data: self.tga.image_data(),
+            x: 0,
+            y: 0,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct ImageBmpIterator<'a, C: 'a>
```
Comment (rfuest) [inline embedded-graphics/src/image/image_tga.rs]: Rename to `ImageTgaIterator`

File: tinytga/src/header.rs [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    /// Image type
+    pub image_type: ImageType,
+
+    /// Color map origin
+    pub color_map_start: u16,
+
+    /// Length of color map
+    pub color_map_len: u16,
+
+    /// Number of bits in each color pallette entry, typically 15, 16, 24, or 32 bits
```
Comment (rfuest) [inline tinytga/src/header.rs]: Typo: palette

File: tinytga/src/header.rs [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    /// Image origin (X)
+    pub x_origin: u16,
+
+    /// Image origin (Y)
+    pub y_origin: u16,
+
+    /// Image width in pixels
+    pub width: u16,
+
+    /// Image heigh in pixels
```
Comment (rfuest) [inline tinytga/src/header.rs]: Typo: height

**Comment by jamwaffles** [comment]
Cheers for picking up those nits. Still a ways to go on this PR though 🙂

**Pull request merged**